ingestion/src/metadata/ingestion/source/database/query_parser_source.py

```diff
@@ -58,8 +61,31 @@ def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnecti
 
     def prepare(self):
         """
-        By default, there's nothing to prepare
+        Fetch queries only from DB that is ingested in OM
         """
+        databases: List[Database] = self.metadata.list_all_entities(
+            Database, ["databaseSchemas"], params={"service": self.config.serviceName}
+        )
+        database_name_list = []
+        schema_name_list = []
+
+        for database in databases:
+            database_name_list.append(database.name.__root__)
+            if self.schema_field and database.databaseSchemas:
+                for schema in database.databaseSchemas.__root__:
+                    schema_name_list.append(schema.name)
+
+        if self.database_field and database_name_list:
+            self.filters += (  # pylint: disable=no-member
+                f" AND {self.database_field} IN ('"
+                + "','".join(database_name_list)
+                + "')"
+            )
+
+        if self.schema_field and schema_name_list:
+            self.filters += (  # pylint: disable=no-member
+                f" AND {self.schema_field} IN ('" + "','".join(schema_name_list) + "')"
```

Comment:
In this approach, we are fetching all the database schema and database names  and creating a filter, that will be applied while fetching query history logs!

Reply:
every usage and lineage source will define the `schema_field` and `database_field` available in the query log table